### DISCUSSION TITLE
Fosite Question - HMAC Strategy Key and Key Rotation

### DISCUSSION THREAD

**Discussion by erictg** [ory/.github#37] [Q&A] [answered]
Are the HMAC Strategy keys just supposed to be RSA keys and how many are recommended? Also should these be generated up front or something that would get updated over time?

**Comment by aeneasr** [comment, marked as answer]
Thank you for the question! Could you please point to the code in question to establish content? Generally HMAC-SHA can be used with both symmetric and asymmetric keys so it really depends on which feature you're using :)

**Reply by erictg** [reply]
@aeneasr this function specifically -- https://github.com/ory/fosite/blob/master/compose/compose_strategy.go#L40

**Reply by aeneasr** [reply]
That expects a regular symmetric secret :)

**Reply by erictg** [reply]
How many should I provide for rotation purposes @aeneasr

**Reply by aeneasr** [reply]
If you rotate (change) a secret add it to the list. If you never changed the secrets, leave it empty

**Reply by erictg** [reply]
@aeneasr Looks like that struct doesn't have built in functionality to add a new one. If I wanted to add one after a rotation, would I move the current global secret to the list of rotated secrets then change the global secret to the new one?

**Reply by aeneasr** [reply]
exactly

**Reply by erictg** [reply]
@aeneasr I submitted a [pull request to fosite](https://github.com/ory/fosite/pull/662) that does this internally :)

**Reply by aeneasr** [reply]
awesome!
